src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -39,28 +45,45 @@ public List<BookDto> findAll(String email, Pageable pageable) {
     }
 
     @Override
-    public BookDto findById(Long id) {
+    public BookResponseDto findById(Long id) {
         Book book = findByIdOrThrowException(id);
         return bookMapper.toDto(book);
     }
 
     @Override
-    public BookDto update(Long id, BookRequestDto requestDto) {
+    public BookResponseDto update(Long id, BookRequestDto requestDto) {
         Book book = findByIdOrThrowException(id);
         bookMapper.updateFromDto(requestDto, book);
+        setCategories(book, requestDto.getCategoryIds());
         Book updatedBook = bookRepository.save(book);
         return bookMapper.toDto(updatedBook);
     }
 
     @Override
-    public BookDto delete(Long id) {
-        Book book = findByIdOrThrowException(id);
-        bookRepository.delete(book);
-        return bookMapper.toDto(book);
+    public void delete(Long id) {
+        if (bookRepository.existsById(id)) {
+            bookRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Book not found");
+        }
     }
 
     private Book findByIdOrThrowException(Long id) {
         return bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
     }
+
+    private void setCategories(Book book, Set<Long> categoryIds) {
+        if (categoryIds != null && !categoryIds.isEmpty()) {
+            Set<Category> categories = categoryIds.stream()
+                    .map(categoryId -> categoryRepository.findById(categoryId)
+                            .orElseThrow(()
+                                    -> new EntityNotFoundException("Category not found with id "
+                                    + categoryId)))
+                    .collect(Collectors.toSet());
+            book.setCategories(categories);

```

Comment:
The performance of this code is very low, because you make N request to DB, please investigate how performance could be improved